### DISCUSSION TITLE
Using Display resolution in an action

### DISCUSSION THREAD

**Discussion by UUoocl** [WarmUpTill/SceneSwitcher#994] [Q&A] [answered]
I would like to use a Display's resolution in an Advance Scene Switcher action. Is it possible to get the display resolution and coordinates using advanced scene switcher?  
The Source "Display Capture" shows the resolution and coordinates in the "Display" name. 
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/fb2da1b2-e159-4849-b3ca-e3b1fd726591)

However, the "Display" settings do not appear in the Advance Scene Switcher Source condition.  
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/324307e8-04d9-4d56-aee2-c1575f04b3bc)

My idea is to use RegEx to store the Resolution and Coordinates in Variables, but can't figure out how.   Is this currently possible or should this become an Issue Request?  Thanks for reading.

**Comment by WarmUpTill** [comment]
I don't think there is a way to get the display size at the moment.
I might add this to the "Display" condition.

Thanks for the suggestion!

**Reply by WarmUpTill** [reply]
A build with the requested functionality will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/7679958103?pr=998

![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/241d1b44-5146-4efb-a69a-c239a01149dc)

**Reply by UUoocl** [reply]
Thank you for considering my question and drafting a solution! Testing this version had mixed results. I'm using a MacBook M1 and some of the resolutions are appearing 2 times larger than the selected size.  For example, 1512 x 945 is appearing as 3024 x 1890.  
The 'Display Capture' Source shows the selected size
<img width="387" alt="image" src="https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/7440ccf5-70ce-4635-9add-d3ad57d90fbe">

The Advanced Scene Switch "Display Condition" shows the size * 2
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/2708e2af-bd33-4b40-9d75-fcca21afabde)

I'm looking to find the Display size and location, then use those values in Advanced Scene Switcher.  My setup has two monitors arranged in a stack.   
As a work around, I'm working on an AppleScript to bring the values into OBS.

**Reply by WarmUpTill** [reply, marked as answer]
Hm .. that seems to be happening because of some [scaling](https://doc.qt.io/qt-5/qscreen.html#devicePixelRatio-prop) done on "retina" display:
> Common values are 1.0 on normal displays and 2.0 on "retina" displays. Higher values are also possible.

I added the option to ignore this scaling or factor it into the reported size.

![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/2f1a9069-034b-45bf-8cea-e33413a9f386)

A new build will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/7681824381?pr=998

**Reply by UUoocl** [reply]
You figured it out! This version works as expected.  Thanks! This is great.
